### PR TITLE
Replace coveralls with codecov

### PR DESCRIPTION
**Pull request by kiudee**
## Description
Remove coveralls from the repository and set up codecov instead.

## Motivation and Context
We currently use Coveralls to report coverage statistics for each new pull request. Sadly, it does not handle parallel builds correctly and thus is quite useless.
Codecov seems to be widely used, better maintained and can be configured in detail.

## How Has This Been Tested?
By creating this pull request

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: docs/intro.rst [hunk 1]
```diff
@@ -41,7 +41,8 @@ As a simple "Hello World!"-example we will try to learn the Pareto problem:
    gen = ChoiceDatasetGenerator(dataset_type='pareto',
                                    n_objects=30,
                                    n_features=2)
-   X_train, Y_train, X_test, Y_test = gen.get_single_train_test_split()                     
+   X_train, Y_train, X_test, Y_test = gen.get_single_train_test_split()
+
```
Comment (timokau) [inline docs/intro.rst]: This seems unrelated, was the change intentional?

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  status:
+    project: yes
+    patch: yes
+    changes: no
+
+comment:
+  layout: "reach, diff, flags, files"
+  behavior: default
+  require_changes: false  # if true: only post the comment if coverage changes
```
Comment (timokau) [inline .codecov.yml]: I like that this will only post a comment [once](https://docs.codecov.io/docs/pull-request-comments) and then update it. I don't think coveralls did that. Still, I would kind of prefer no comment at all. It shows up in githubs CI listing anyway, what value does the additional comment add?

Comment (kiudee) [reply .codecov.yml]: I think it could be useful to see at a glance how the coverage for a pull request changed. Sure it’s available with two clicks in the CI listing, but the information is more scattered on the codecov website.
Additionally, I think it is easy to ignore in all other cases, where it is not needed.

File: docs/intro.rst (hunk 1 above)
Comment (kiudee) [reply docs/intro.rst]: That is unrelated indeed. But we should fix it separately and get a proper docs linter up and running.

File: .codecov.yml (hunk 2 above)
Comment (timokau) [reply .codecov.yml]: I don't think the usefulness justifies the verbosity. Its only really relevant when the coverage meaningfully decreases, which hopefully will be the minority of the time (and then the big red x should be obvious enough).

I made the comment at `require_changes` because I think `require_changes = true` would be a nice compromise if you prefer to keep the comment. I don't have a very strong preference either way :)

Comment (kiudee) [reply .codecov.yml]: I can see the concern, which is why I would propose monitoring the usefulness over time. I will merge it as is for now.
